### ISSUE TITLE
Slow loading time for the example shell app on CloudFront and locally

### ISSUE THREAD

**Issue by diandsonc** [angular-architects/module-federation-plugin#679]
### For which library do you need help?

native-federation

### Question

Hi,
I've been working with the base Angular project provided in the repo, and I deployed the example shell app to CloudFront. However, it takes a little over 8 seconds to load the site, and even when running locally, there's a white screen for a few seconds before the content loads. This issue doesn't occur when using **module-federation**.

Is there any way to optimize or correct this behavior? Here's some additional information:

- **Setup:** I followed the steps outlined in the repository for the base project.
- **Issue on CloudFront:** Takes around 8+ seconds for the content to appear.
- **Local Issue:** Even locally, there's a noticeable delay with a blank screen for a few seconds before the content loads fully.
- **Comparison with module-federation:** With module-federation, I don’t experience this lag either locally or in production.
- **Expected Behavior:** Content should load faster without a noticeable blank screen.

Is there any optimization or configuration I should be aware of to fix this? Appreciate any help or guidance!

Thanks in advance!

**Comment by rainerhahnekamp** [comment]
Usually native federation compiles the shared libraries but if you use the build, this should already have happened before. It sounds like CloudFronend runs that build as well? Do you have any logs, etc?

**Comment by manfredsteyer** [comment]
Hi @diandsonc,

can you please share the information from your network tab in the dev tools?

**Comment by diandsonc** [comment]
Hi @manfredsteyer 

Here are the screenshots from my network tab. You can see the loading times for various JS files. The page is taking around 8 seconds to load, as reflected in the waterfall chart. The files hosted on CloudFront seem to load sequentially, but there’s a delay before they start fetching, even for local setup.

Let me know if you spot anything that might be causing the delay or if you need any more details!


![net](https://github.com/user-attachments/assets/92bf61e5-4efb-4b0b-b566-e09940e2b595)
![net continue](https://github.com/user-attachments/assets/38e39760-0bbf-49fb-a016-81579e15dab4)
![files](https://github.com/user-attachments/assets/37ee27e6-ee11-4545-9ca3-69ea3fb4e6ba)

**Comment by diandsonc** [comment]
> Hi @diandsonc,
> 
> can you please share the information from your network tab in the dev tools?

Hi @manfredsteyer,

I have two screenshots from the network tab: one from the base project and one from my project. I also noticed that when working with environments, the environment file sometimes loads in a different stage, and variables that depend on it are called before it's fully loaded, causing errors. This could be affecting the load time.

I'm sharing the screenshots for you to take a look. For my project It might be related to the environment loading issue. What do you think?

Base project:
![net_shell](https://github.com/user-attachments/assets/cc9c64e3-f33b-4ba3-ae21-22a3831448a9)

My project
![net_project](https://github.com/user-attachments/assets/1851ee16-eb27-4d3d-a99d-58ba8cf53c37)



> Usually native federation compiles the shared libraries but if you use the build, this should already have happened before. It sounds like CloudFronend runs that build as well? Do you have any logs, etc?

@rainerhahnekamp I'm trying to understand what might be causing this

**Comment by diandsonc** [comment]
Hi @manfredsteyer , @rainerhahnekamp 

I managed to partially solve the issue. The initial load is still a bit slow but acceptable now. The main problem was with my caching policy, somehow the cache wasn’t being generated properly. After fixing that, the load times improved significantly.

I also adjusted the response header policy, which was causing some other issues that were slowing things down. Lastly, enabling HTTP/3 made a noticeable difference in performance as well.

Thanks
